backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/types/dsl/ElasticsearchIndexFieldTypeFactoryContext.java

```diff
@@ -14,6 +14,33 @@
  */
 public interface ElasticsearchIndexFieldTypeFactoryContext extends IndexFieldTypeFactoryContext {
 
-	ElasticsearchJsonStringIndexFieldTypeContext<?> asJsonString(String mappingJsonString);
+	/**
+	 * Define a native field type.
+	 * <p>
+	 * A native field type has the following characteristics:
+	 * <ul>
+	 *     <li>Hibernate Search doesn't know its exact type, so it must be entirely defined as a JSON object,
+	 *     provided as the {@code mappingJsonString} parameter</li>
+	 *     <li>When indexing, fields of this type must be populated with JSON.
+	 *     The field has a string type, but the string is interpreted as JSON, so it can contain a boolean, an integer,
+	 *     or even an object.</li>
+	 *     <li>The predicate/sort/projection DSLs have only limited support for fields of this type.
+	 *     Some features may not work and throw an exception, such as phrase predicates.
+	 *     It is recommended to create the predicate/sort/projections targeting these fields from JSON
+	 *     using {@link org.hibernate.search.backend.elasticsearch.search.dsl.predicate.ElasticsearchSearchPredicateFactoryContext#fromJson(String)}
+	 *     or {@link org.hibernate.search.backend.elasticsearch.search.dsl.sort.ElasticsearchSearchSortContainerContext#fromJson(String)}</li>
+	 * </ul>
+	 *
+	 * @param jsonString A string representing an Elasticsearch field mapping as a JSON object.
+	 * The JSON object must be a syntactically correct Elasticsearch field mapping.
+	 * See <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping.html">the Elasticsearch documentation</a>.
+	 * @return A context allowing to get the resulting predicate.
+	 */
+	/**
+	 *
+	 * @param mappingJsonString
+	 * @return
+	 */
```

Comment:
There is an issue with the javadoc. You have 2 javadoc blocks and in the first one `jsonString` should be `mappingJsonString`.